Wikipedia/Code/MWKDataStore.m

```diff
@@ -114,25 +122,173 @@ - (void)dealloc {
 }
 
 - (instancetype)init {
-    self = [self initWithDatabase:[YapDatabase sharedInstance]];
+    self = [self initWithContainerURL:[[NSFileManager defaultManager] wmf_containerURL]];
     return self;
 }
 
-- (instancetype)initWithDatabase:(YapDatabase *)database {
-    self = [self initWithDatabase:database legacyDataBasePath:[[MWKDataStore class] mainDataStorePath]];
-    return self;
+static pid_t currentPid() {
```

Comment:
Actually looking through, it seems like not. You really only want to process the notification once per process not once per instance